### ISSUE TITLE
Dependency Dashboard Undoes Approvals

### ISSUE THREAD

**Issue by Kurt-von-Laven** [renovatebot/renovate#18614]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

[ScribeMD](https://github.com/ScribeMD) recently installed Forking Renovate on all of our public repositories using [this shared config](https://github.com/ScribeMD/.github/blob/0.4.1/default.json). The dependency dashboard issues were immediately opened correctly on every repository. Anytime we approve a branch by checking the box, the bot correctly moves that branch to the "Other Branches" section the next time it runs. However, the next time it runs after that, it moves the branch back to the "Pending Approvals" section. Hence, Renovate never opens any pull requests. See the revision history of ScribeMD/template-repo#16 for example.

### Relevant debug logs

When I log into Renovate's app dashboard, it doesn't display any repos. I assume this is either because we are using Forking Renovate or because no Renovate jobs have run.

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by viceice** [comment]
post upgrade tasks are not supported on hosted app

**Comment by Kurt-von-Laven** [comment]
Thank you for the quick reply and for pointing this out. I have made the corresponding change to our base config, but continue to observe the same symptom in ScribeMD/.github#96, so if that was the cause of this behavior, it was not the only cause. I have not propagated the config change to our other repositories for this reason. Is there any way for me to access some logs to aid in troubleshooting given that we are using Forking Renovate?

**Comment by rarkins** [comment]
Unfortunately logs are not available in the dashboard for forking renovate, but I can access them manually. I picked the largest log out of recent ones and found this error:

```json
{
  "level": 20,
  "branch": "renovate/lock-file-maintenance",
  "err": {
    "name": "HTTPError",
    "code": "ERR_NON_2XX_3XX_RESPONSE",
    "timings": {
      "start": 1666909780724,
      "socket": 1666909780725,
      "lookup": 1666909780734,
      "connect": 1666909780734,
      "secureConnect": 1666909780748,
      "upload": 1666909780748,
      "response": 1666909780902,
      "end": 1666909780904,
      "phases": {
        "wait": 1,
        "dns": 9,
        "tcp": 0,
        "tls": 14,
        "request": 0,
        "firstByte": 154,
        "download": 2,
        "total": 180
      }
    },
    "message": "Response code 403 (Forbidden)",
    "stack": "HTTPError: Response code 403 (Forbidden)\n    at Request.<anonymous> (/home/ubuntu/renovateapp/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)",
    "options": {
      "headers": {
        "user-agent": "Renovate Bot (GitHub App 2740)",
        "accept": "application/vnd.github.machine-man-preview+json",
        "authorization": "***********",
        "content-type": "application/json",
        "content-length": "2313",
        "accept-encoding": "gzip, deflate, br"
      },
      "url": "https://api.github.com/repos/renovate-bot/.github-6/git/trees",
      "hostType": "github",
      "username": "",
      "password": "",
      "method": "POST",
      "http2": false
    },
    "response": {
      "statusCode": 403,
      "statusMessage": "Forbidden",
      "body": {
        "message": "Resource not accessible by integration",
        "documentation_url": "https://docs.github.com/rest/reference/git#create-a-tree"
      },
      "headers": {
        "server": "GitHub.com",
        "date": "Thu, 27 Oct 2022 22:29:40 GMT",
        "content-type": "application/json; charset=utf-8",
        "transfer-encoding": "chunked",
        "x-github-media-type": "github.v3; param=machine-man-preview; format=json",
        "x-ratelimit-limit": "5000",
        "x-ratelimit-remaining": "4992",
        "x-ratelimit-reset": "1666913106",
        "x-ratelimit-used": "8",
        "x-ratelimit-resource": "core",
        "access-control-expose-headers": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset",
        "access-control-allow-origin": "*",
        "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
        "x-frame-options": "deny",
        "x-content-type-options": "nosniff",
        "x-xss-protection": "0",
        "referrer-policy": "origin-when-cross-origin, strict-origin-when-cross-origin",
        "content-security-policy": "default-src 'none'",
        "vary": "Accept-Encoding, Accept, X-Requested-With",
        "content-encoding": "gzip",
        "x-github-request-id": "82EA:1E0F:460B9F8:47DD9DF:635B0654",
        "connection": "close"
      },
      "httpVersion": "1.1",
      "retryCount": 0
    }
  },
  "msg": "GitHub failure: Resource not accessible by integration",
  "time": "2022-10-27T22:29:40.907Z"
}
```

I don't yet know what would be causing that or if it's the only root cause.

I would also suggest experimenting with a different branch to lock file maintenance, because it has some peculiarities which make it harder to predict.

**Comment by Kurt-von-Laven** [comment]
Thanks for digging this up. At face value that sounds like the token Forking Renovate is using to update its own fork lacks the `contents:write` permission, but I know it isn't that simple. Do the logs contain the request body that received a 403 response? The symptom is the same with every dependency I have tried. I just made a few attempts at bumping Python to 3.11.0 that behaved identically.

**Comment by rarkins** [comment]
No, we don't log request body.

The Forking Renovate bot actually has two tokens - one for `@renovate-bot` and one for `@forking-renovate[bot]`.

I suggest you should try to reduce the complexity so we can troubleshoot more easily.

First of all, uninstall current repos if you can, to reduce the chance we accidentally create noise in them.

Then, focus on a single test repo and do not use features like approval. Keep the config minimal, and the dependencies too (e.g. one dependency may be enough).

**Comment by Kurt-von-Laven** [comment]
It seems from my failed attempts to create a minimal reproduction that this bug is in some way related to state retained by the Forking Renovate bot. Note that the Dependency Dashboard approval issue is still around on [this reproduction](https://github.com/Kurt-von-Laven/.github) despite my having incrementally reduced it to [a minimal config](https://github.com/Kurt-von-Laven/.github/blob/38597808431c7da7a1cdd669df419c9a274c594e/default.json). Given this, should I be uninstalling Forking Renovate, deleting the repository, recreating the repository with a different config/dependencies, and reinstalling Forking Renovate between each experiment? Just to be clear, Kurt-von-Laven/.github is the reproduction, and ScribeMD/.github is the production repository I based it on.

**Comment by Kurt-von-Laven** [comment]
It's [`platformCommit: true`](https://github.com/Kurt-von-Laven/minimal-repro/commit/3d3fba8e47041b9cdc7d547fee1e6264feb4e13e). I'm guessing it works with Mend Renovate but not Forking Renovate. I essentially ran the same experiments in reverse, starting with all of the dependencies but a minimal config and incrementally adding settings from the production repository. https://github.com/Kurt-von-Laven/minimal-repro worked fine until I added `{platformCommit: true}`. I don't know how to fix this issue in the production repository since my previous experiment already demonstrated that removing `platformCommit: true` is not sufficient to undo the damage. I can try uninstalling and reinstalling Forking Renovate if you think that's wise, but I want to make sure that wouldn't wipe out the logs for you all.

**Comment by rarkins** [comment]
Good catch. I don't think `platformCommit` can work currently because the app does not have permissions to commit to the forked repo.

I have now manually deleted the caches of both your real and test `.github` repos. Is that what you needed?

**Comment by Kurt-von-Laven** [comment]
Thank you! Yes, that appears to have resolved the issue. Oh, I see. When I enabled that feature, I was expecting Forking Renovate to perform a platform commit on the fork rather than the repo it doesn't have access to, but maybe that isn't desirable?

**Comment by rarkins** [comment]
To do that the app token would need to have write permissions to the fork repo, which it doesn't. We're brainstorming on whether we could have a solution with *two* app tokens - one for the upstream target repo and one for the downstream fork repo, so that then the platform commit would work.

**Comment by Kurt-von-Laven** [comment]
Oh, I see. Yes, such a solution would have matched my expectations regarding this features' behavior from reading the docs.
